presto-spi/src/main/java/com/facebook/presto/spi/Plugin.java

```diff
@@ -13,12 +13,25 @@
  */
 package com.facebook.presto.spi;
 
+import com.facebook.presto.spi.block.BlockEncodingSerde;
+import com.facebook.presto.spi.type.TypeManager;
+
 import java.util.List;
 import java.util.Map;
 
 public interface Plugin
 {
     default void setOptionalConfig(Map<String, String> optionalConfig) {}
 
+    default void setTypeManager(TypeManager typeManager) {}
+
+    default void setNodeManager(NodeManager nodeManager) {}
+
+    default void setBlockEncodingSerde(BlockEncodingSerde blockEncodingSerde) {}
```

Comment:
It doesn't seem like a good idea to make all plugin types aware of some of these objects. For example, why would an event listener or auth plugin care about a page sorter?